x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -268,6 +271,97 @@ public void testHighWatermarkCanBeExceededOnColdOrFrozenNode() throws Exception
         });
     }
 
+    public void testOvercommitInitializingSearchableSnapshotShards() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startNode(onlyRole(DATA_HOT_NODE_ROLE));
+
+        int nbIndices = createIndices();
+
+        String repository = "repository";
+        assertAcked(
+            client().admin()
+                .cluster()
+                .preparePutRepository(repository)
+                .setType("mock")
+                .setSettings(Settings.builder().put("location", randomRepoPath()).build())
+        );
+
+        String snapshotName = "snapshot";
+        createSnapshot(repository, snapshotName, nbIndices);
+
+        Map<String, Long> indicesStoresSizes = sizeOfShardsStores("index-*");
+        assertAcked(client().admin().indices().prepareDelete("index-*"));
+
+        String otherDataNode = internalCluster().startNode(
+            Settings.builder().put(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), DiscoveryNodeRole.DATA_COLD_NODE_ROLE.roleName()).build()
+        );
+        ensureStableCluster(3);
+
+        String otherDataNodeId = client().admin().cluster().prepareState().get().getState().nodes().resolveNode(otherDataNode).getId();
+        logger.info(
+            "--> reducing disk size of node [{}/{}] so that all shards except one can fit on the node",
+            otherDataNode,
+            otherDataNodeId
+        );
+        String indexToSkip = randomFrom(indicesStoresSizes.keySet());
+        Map<String, Long> indicesToBeMounted = indicesStoresSizes.entrySet()
+            .stream()
+            .filter(e -> e.getKey().equals(indexToSkip) == false)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        long totalSpace = indicesToBeMounted.values().stream().mapToLong(e -> e).sum() + WATERMARK_BYTES + 1024L;
+        getTestFileStore(otherDataNode).setTotalSpace(totalSpace);
+
+        logger.info("--> refreshing cluster info");
+        InternalClusterInfoService masterInfoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(
+            ClusterInfoService.class
+        );
+        ClusterInfoServiceUtils.refresh(masterInfoService);
+        assertThat(
+            masterInfoService.getClusterInfo().getNodeMostAvailableDiskUsages().get(otherDataNodeId).getTotalBytes(),
+            equalTo(totalSpace)
+        );
+
+        var mockRepository = (MockRepository) internalCluster().getCurrentMasterNodeInstance(RepositoriesService.class)
+            .repository(repository);
+        // Prevent searchable snapshot shards from quickly jumping from INITIALIZED to STARTED
+        mockRepository.setBlockOnceOnReadSnapshotInfoIfAlreadyBlocked();
+
+        String prefix = "mounted-";
+        mountIndices(indicesToBeMounted.keySet(), prefix, repository, snapshotName, FULL_COPY);
+        mockRepository.unblock();
```

Comment:
I am not sure I understand why this unblocks here. Should it not wait until after we mounted the `indexToSkip`? Otherwise the test might only be checking against started indices.

Reply:
It's a good point. I actually not sure whether we need to unblock the repository at all.